pkg/validation/attributes/attributes_command_test.go

```diff
@@ -0,0 +1,172 @@
+package attributes
+
+import (
+	"testing"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateExecCommand(t *testing.T) {
+
+	tests := []struct {
+		name       string
+		testFile   string
+		expected   v1alpha2.ExecCommand
+		attributes apiAttributes.Attributes
+		wantErr    bool
+	}{
+		{
+			name:     "Good Substitution",
+			testFile: "test-fixtures/commands/exec.yaml",
+			expected: v1alpha2.ExecCommand{
+				CommandLine: "tail -f /dev/null",
+				WorkingDir:  "FOO",
+				Component:   "BAR",
+				LabeledCommand: v1alpha2.LabeledCommand{
+					Label: "1",
+				},
+				Env: []v1alpha2.EnvVar{
+					{
+						Name:  "FOO",
+						Value: "BAR",
+					},
+				},
+			},
+			attributes: apiAttributes.Attributes{}.FromMap(map[string]interface{}{
+				"version": "1",
+				"devnull": "/dev/null",
+				"bar":     "BAR",
+				"foo":     "FOO",
+			}, nil),
+			wantErr: false,
+		},
+		{
+			name:     "Invalid Reference",
+			testFile: "test-fixtures/commands/exec.yaml",
+			attributes: apiAttributes.Attributes{}.FromMap(map[string]interface{}{
+				"foo": "FOO",
+			}, nil),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testExecCommand := v1alpha2.ExecCommand{}
+
+			readFileToStruct(t, tt.testFile, &testExecCommand)
+
+			err := validateExecCommand(tt.attributes, &testExecCommand)
+			if tt.wantErr == (err == nil) {
+				t.Errorf("error: %v", err)
```

Comment:
This isn't really descriptive enough IMO -- you could have a test fail with `error: <nil>` which could be frustrating. There's no real need for terseness here:
```go
if tt.wantErr && err == nil {
  t.Errorf("Expected error from test but got nil")
} else if !tt.wantErr && err != nil {
  t.Errorf("Got unexpected error: %s", err)
}
```

Reply:
Addressed